src/rsz/test/cpp/TestBufferRemoval.cc

```diff
@@ -0,0 +1,225 @@
+// Copyright 2023 Google LLC
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#include <tcl.h>
+#include <unistd.h>
+
+#include <array>
+#include <filesystem>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <set>
+
+#include "AbstractSteinerRenderer.h"
+#include "db_sta/MakeDbSta.hh"
+#include "db_sta/dbNetwork.hh"
+#include "db_sta/dbSta.hh"
+#include "dpl/MakeOpendp.h"
+#include "gmock/gmock.h"
+#include "grt/GlobalRouter.h"
+#include "gtest/gtest.h"
+#include "odb/lefin.h"
+#include "rsz/MakeResizer.hh"
+#include "rsz/Resizer.hh"
+#include "sta/Corner.hh"
+#include "sta/FuncExpr.hh"
+#include "sta/Graph.hh"
+#include "sta/Liberty.hh"
+#include "sta/PathAnalysisPt.hh"
+#include "sta/Search.hh"
+#include "sta/Sta.hh"
+#include "sta/Units.hh"
+#include "stt/SteinerTreeBuilder.h"
+#include "utl/Logger.h"
+#include "utl/deleter.h"
+
+namespace rsz {
+
+std::once_flag init_sta_flag;
+
+class BufRemTest : public ::testing::Test
+{
+ protected:
+  void SetUp() override
+  {
+    // this will be so much easier with read_def
+    db_ = utl::deleted_unique_ptr<odb::dbDatabase>(odb::dbDatabase::create(),
+                                                   &odb::dbDatabase::destroy);
+    std::call_once(init_sta_flag, []() { sta::initSta(); });
+    sta_ = std::unique_ptr<sta::dbSta>(ord::makeDbSta());
+    sta_->initVars(Tcl_CreateInterp(), db_.get(), &logger_);
+    auto path = std::filesystem::canonical("./Nangate45/Nangate45_typ.lib");
+    library_ = sta_->readLiberty(path.string().c_str(),
+                                 sta_->findCorner("default"),
+                                 sta::MinMaxAll::all(),
+                                 /*infer_latches=*/false);
+    odb::lefin lefParser(
+        db_.get(), &logger_, /*ignore_non_routing_layers*/ false);
+    const char* libName = "Nangate45.lef";
+    odb::dbLib* dbLib = lefParser.createTechAndLib(
+        "tech", libName, "./Nangate45/Nangate45.lef");
+
+    sta_->postReadLef(/*tech=*/nullptr, dbLib);
+
+    sta::Units* units = library_->units();
+    power_unit_ = units->powerUnit();
+    db_network_ = sta_->getDbNetwork();
+
+    // create a chain consisting of 4 buffers
+    odb::dbChip* chip = odb::dbChip::create(db_.get());
+    odb::dbBlock* block = odb::dbBlock::create(chip, "top");
+    db_network_->setBlock(block);
+    block->setDieArea(odb::Rect(0, 0, 1000, 1000));
+    // register proper callbacks for timer like read_def
+    sta_->postReadDef(block);
```

Comment:
I would say this stuff would usually go in the test itself unless you plan to use this same chain of buffers in multiple tests.

Reply:
Yes, I want to use the same design for multiple tests.